### PR TITLE
Implement `astropy.table.Row.get()`

### PR DESCRIPTION
**Pull request by eerovaher**
### Description

Currently code that tries to access a column in a `Row` when the presence of the column is not guaranteed either has to check for the presence of the column beforehand or handle the resulting `KeyError` explicitly. In many cases (see e.g. https://github.com/astropy/astropy/pull/14780#discussion_r1200857265) it might be appealing to use `Row.columns.get()` instead, which allows specifying a fallback value to be used if the column is not present, but (surprisingly) `Row.columns` actually returns the `columns` from the parent table of the `Row`: https://github.com/astropy/astropy/blob/88790514bdf248e43c2fb15ee18cfd3390846145/astropy/table/row.py#L152-L154

Changing the behavior of `Row.columns` would be a backwards-incompatible change, so it is safer to implement a new `Row.get()` method.

~~When implementing the unit tests for the new feature I thought that it would be better if the test table would be a module level fixture so that it would not be recreated for every instance of the parametrized tests, and then I found it useful to share the test table with an already existing test so that it could be reused even more. Because of this I also refactored and parametrized an already existing unit test for `Row`.~~

### CODE REVIEW

**Comment by eerovaher** [comment]
I had to force-push the second commit because an issue and a pull request were opened while I was writing the opening message, so I had to update the change log entry filename to match the actual pull request number.

File: astropy/table/tests/test_row.py [hunk 1]
```diff
@@ -355,20 +361,35 @@ def test_uint_indexing():
...
+
+@pytest.mark.parametrize(
+    "column,expectation",
+    [
+        pytest.param("b", np.True_, id="column_present"),
+        pytest.param("y", False, id="column_not_present"),
+    ],
+)
+def test_row_get_with_specified_default(column, expectation, small_table):
+    assert small_table[1].get(column, False) is expectation
```
Comment (taldcroft) [inline astropy/table/tests/test_row.py:395]: While I understand and appreciate the idea of atomic tests, this just seems overboard. Keeping in mind "Simple is better than complex", what about a simple version that is much faster for people to read and confirm that it has the right behavior. It might also be just as fast to run in tests because there must be a substantial overhead in pytest to generate and collect 4 different tests.
```
def test_row_get():
    t = table.Table([[1, 2]], names=["a"])
    assert t[0].get("a") == 1
    assert t[0].get("x") is None
    assert t[1].get("a", False) == 1
    assert t[1].get("x", False) is False
```

File: astropy/table/tests/test_row.py [hunk 2]
```diff
@@ -343,7 +348,8 @@ def test_row_tuple_column_slice_transaction():
     assert t[1] == tc[1]
 
 
-def test_uint_indexing():
+@pytest.mark.parametrize("dtype", [int, np.int_, np.uint])
+def test_uint_indexing(dtype, small_table):
```
Comment (taldcroft) [inline astropy/table/tests/test_row.py:352]: I'm not convinced there is a substantial performance improvement in the end due to overhead in collecting three pytests tests instead of one and using a fixture. Probably better to just revert this unrelated change.

**Review by taldcroft — CHANGES_REQUESTED** [review]
Thanks @eerovaher! 

The actual for the `get()` method looks good. The test seems overly complex for something extremely simple. I'd also suggest adding an example or two in
https://docs.astropy.org/en/stable/table/access_table.html#basics and/or https://docs.astropy.org/en/stable/table/access_table.html#accessing-data

**Comment by mhvk** [comment]
From the sidelines: Definitely happy with making `Row` a bit more `dict`-like. I was wondering if `Table` should similarly get a `get` method, but I guess having to pass in a default column makes less sense.

File: astropy/table/tests/test_row.py (hunk 1 above)
Comment (eerovaher) [reply astropy/table/tests/test_row.py]: > simple version that is much faster for people to read

In my opinion well parametrized tests with detailed names make the output of `pytest` much more readable, so that the purpose of a test might even be understood without having to read the source code at all. But in this case the test is indeed very short and simple so that even if it does fail the entirety of the test code is easily readable from the traceback `pytest` prints to the terminal. Not parametrizing this test therefore does not hurt the usability of `pytest` and it does shorten the source code.

File: astropy/table/tests/test_row.py (hunk 2 above)
Comment (eerovaher) [reply astropy/table/tests/test_row.py]: I think that parametrizing this test would make the output of `pytest` more user-friendly, but having changed the implementation of the `Row.get()` unit tests I can't justify making this change in this pull request.

**Comment by eerovaher** [comment]
> I was wondering if `Table` should similarly get a `get` method

There would also be the question if an integer key like `t.get(0)` should return a row.

`Table.columns.get()` already works and is in my opinion simple enough to use and simple enough to understand if it is encountered in code. The main motivation for implementing a separate `Row.get()` was that `Row.columns.get()` does not produce the expected outcome.

**Review by taldcroft — APPROVED** [review]
Awesome, thanks @eerovaher !
